sdk/core/azure-core/inc/azure/core/internal/strings.hpp

```diff
@@ -30,6 +30,22 @@ namespace Azure { namespace Core { namespace _internal {
       return (c < 'A' || c > 'Z') ? c : c + ('a' - 'A');
     }
 
+    static constexpr bool IsDigit(char c) noexcept { return c >= '0' && c <= '9'; }
+
+    static constexpr bool IsHexDigit(char c) noexcept
+    {
+      return IsDigit(c) || (c >= 'A' && c <= 'F') || (c >= 'a' && c <= 'f');
+    }
+
+    static constexpr bool IsAlphaNumeric(char c) noexcept
+    {
+      return IsDigit(c) || (c >= 'A' && c <= 'Z') || (c >= 'a' && c <= 'z');
+    }
+
+    static constexpr bool IsSpace(char c) noexcept { return c == ' ' || (c >= '\t' && c <= '\r'); }
+
+    static constexpr bool IsPrintable(char c) noexcept { return c >= ' ' && c <= '~'; }
```

Comment:
Similarly here, I don't see this to be generally useful within the SDK. It only has one caller, in `curl.cpp` in `DumpCurlInfoToLog` for filtering characters before logging. This can be moved into that file as an implementation detail, until we see any other use case for it.

Reply:
This is internal class. And I think that once the code needs to use `isblank()` or `isgraph()` the right way to evolve the code is to add a corresponding method here, and not to create a local copy.